src/accelerate/accelerator.py

```diff
@@ -357,6 +447,7 @@ def free_memory(self):
         method between two trainings with different models/optimizers.
         """
         self._optimizers = []
+        self.deepspeed_engine = None
```

Comment:
Is this the only reference we create? It seems so but I want to be certain.

Reply:
yes